src/Controls/src/BindingSourceGen/Setter.cs

```diff
@@ -38,16 +39,23 @@ void AddPatternMatchingExpression(string pattern)
                     AddPatternMatchingExpression("{}");
                 }
 
-                accessAccumulator = AccessExpressionBuilder.ExtendExpression(accessAccumulator, innerPart);
+                accessAccumulator = AccessExpressionBuilder.ExtendExpression(accessAccumulator, innerPart, id: id);
             }
             else
             {
-                accessAccumulator = AccessExpressionBuilder.ExtendExpression(accessAccumulator, part);
+                accessAccumulator = AccessExpressionBuilder.ExtendExpression(accessAccumulator, part, id, part == path.Last());
             }
         }
 
         return new Setter(
             patternMatchingExpressions.ToArray(),
-            AssignmentStatement: $"{accessAccumulator} = {assignedValueExpression};");
+            AssignmentStatement: BuildAssignmentStatement(accessAccumulator, path.Any() ? path.Last() : null, id, assignedValueExpression));
     }
+
+    public static string BuildAssignmentStatement(string accessAccumulator, IPathPart? lastPart, uint id, string assignedValueExpression = "value") =>
+        lastPart switch
+        {
+            InaccessibleMemberAccess inaccessibleMemberAccess when inaccessibleMemberAccess.Kind == AccessorKind.Property => $"SetUnsafeProperty{id}{inaccessibleMemberAccess.MemberName}({accessAccumulator}, {assignedValueExpression});",
+            _ => $"{accessAccumulator} = {assignedValueExpression};",
```

Comment:
Does `AccessorKind.Property` need special handling too?

Reply:
This handles `AccessorKind.Property`, am I missing something?

Reply:
oh I meant `.Field`, sorry for the confusion 😄

Reply:
We have integration tests covering this nicely so we don't really need this special handling for fields